### ISSUE TITLE
Dyninst errors when building with boost-1.69.0

### ISSUE THREAD

**Issue by jgalarowicz** [dyninst/dyninst#526]
Hi all,
Seems like the newer versions of boost cause compiler errors in dyninst.  These errors are from a build with boost-1.69.0 on a Power 9 system at LLNL:

```
==> boost is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/boost-1.69.0-vgp3qovtkp7ngtayfha6nejcuf4jwmp6
==> openssl is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/openssl-1.1.1-26xn4vwusnaqec6zufpwuag4foysfapt
==> cmake is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/cmake-3.13.2-g5hekfpsesidm27ootdxoi5vkdkzing4
==> mrnet is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/mrnet-5.0.1-3-fkbl3phwjclquatmbtjl25hcf7l3fhtp
==> xerces-c is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/xerces-c-3.2.2-h2i54r7a277o56hednbqowb5n7xpxwbt
==> cbtf is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/cbtf-develop-x6ooe4tyvm6d4iivraef35ye6mwnke45
==> elfutils is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/elfutils-0.174-g64svt7zyoch67qabjhjknu4ztnrhvu4
==> intel-tbb is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/intel-tbb-2019.2-sfwpny35v2preom52ow3sghnemo52l25
==> libiberty is already installed in /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/libiberty-2.31.1-adalvkmwug4xjmfwdzzi4oqgll4n54un
==> Installing dyninst
==> Searching for binary cache of dyninst
==> Finding buildcaches in file:/usr/global/tools/openspeedshop/oss-dev/spack-mirror-new/build_cache
==> Finding buildcaches in /usr/global/tools/openspeedshop/oss-dev/cts1/spack-mirror/build_cache
==> No binary for dyninst found: installing from source
==> Fetching file:///usr/global/tools/openspeedshop/oss-dev/cts1/spack-mirror/dyninst/dyninst-develop.tar.gz
######################################################################## 100.0%
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> Staging archive: /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst-develop.tar.gz
==> Created stage in /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h

==> Created stage in /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h
==> No patches needed for dyninst
==> Building dyninst [CMakePackage]
==> Executing phase: 'cmake'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j128'

10 errors found in build log:
     1291    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/LoopAnalyzer.C:39:0:
     1292    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFG.h:708:21: warning: unused parameter 'code
             reg' [-Wunused-parameter]
     1293             CodeRegion *codereg) const
     1294                         ^
     1295    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Sawyer.C:9:0
             :
     1296    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.h: In member function 'boo
             l Sawyer::Message::ColorSpec::isDefault() const':
  >> 1297    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.h:389:96: error: cannot co
             nvert 'boost::logic::tribool' to 'bool' in return
     1298         bool isDefault() const { return COLOR_DEFAULT==foreground && COLOR_DEFAULT==background && !bold; }
     1299                                                                                                    ^
     1300    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Assert.C:9:0
             :
     1301    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.h: In member function 'boo
             l Sawyer::Message::ColorSpec::isDefault() const':
  >> 1302    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.h:389:96: error: cannot co
             nvert 'boost::logic::tribool' to 'bool' in return
     1303         bool isDefault() const { return COLOR_DEFAULT==foreground && COLOR_DEFAULT==background && !bold; }
     1304                                                                                                    ^
     1305    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/common/src/arch.h:40:0,
     1306                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/Parser-speculative.C:
             36:
     1307    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/common/src/arch-power.h:806:39: warning: unused paramete
             r 'mode_64' [-Wunused-parameter]
     1308         instruction(const void *ptr, bool mode_64) {

     ...

     1324                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/JumpTableFormatPred.C
             :1:
     1325    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFG.h:404:15: warning: 'Dyninst::ParseAPI::Pa
             rser' is already a friend of 'Dyninst::ParseAPI::Block'
     1326      friend class Parser;
     1327                   ^
     1328    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.C:8:
             0:
     1329    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.h: In member function 'boo
             l Sawyer::Message::ColorSpec::isDefault() const':
  >> 1330    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.h:389:96: error: cannot co
             nvert 'boost::logic::tribool' to 'bool' in return
     1331         bool isDefault() const { return COLOR_DEFAULT==foreground && COLOR_DEFAULT==background && !bold; }
     1332                                                                                                    ^
     1333    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/JumpTableFormatPred.h
             :4:0,
     1334                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/JumpTableFormatPred.C
             :1:
     1335    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFG.h:708:21: warning: unused parameter 'code
             reg' [-Wunused-parameter]
     1336             CodeRegion *codereg) const
     1337                         ^
     1338    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.C: In member function 'voi
             d Sawyer::Message::StreamBuf::bake()':
  >> 1339    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/util/Message.C:1126:28: error: cannot c
             onvert 'boost::logic::tribool' to 'bool' in assignment
     1340                 anyUnbuffered_ = !bi->second.isBuffered;
     1341                                ^
     1342    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/intel-tbb-2019.2-sfwpny35v2preom52ow3sghnemo52l25/include/tbb/tbb_m
             achine.h:243:0,
     1343                     from /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/intel-tbb-2019.2-sfwpny35v2preom52ow3sghnemo52l25/include/tbb/spin_
             rw_mutex.h:25,
     1344                     from /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/intel-tbb-2019.2-sfwpny35v2preom52ow3sghnemo52l25/include/tbb/concu
             rrent_hash_map.h:32,
     1345                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CodeSource.h:47,

     ...

     1349                              , "+m"(* (int32_t*) ptr)        /* redundant with "memory" */
     1350                                                  ^
     1351    /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/intel-tbb-2019.2-sfwpny35v2preom52ow3sghnemo52l25/include/tbb/machine/mac_ppc.h: In funct
             ion 'int64_t __TBB_machine_cmpswp8(volatile void*, int64_t, int64_t)':
     1352    /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/intel-tbb-2019.2-sfwpny35v2preom52ow3sghnemo52l25/include/tbb/machine/mac_ppc.h:116:46: w
             arning: cast from type 'volatile void*' to type 'int64_t* {aka long int*}' casts away qualifiers [-Wcast-qual]
     1353                              , "+m"(* (int64_t*) ptr)        /* redundant with "memory" */
     1354                                                  ^
  >> 1355    make[2]: *** [parseAPI/CMakeFiles/parseAPI.dir/build.make:768: parseAPI/CMakeFiles/parseAPI.dir/__/dataflowAPI/rose/util/Sawyer.C.o] Error 1
     1356    make[2]: *** Waiting for unfinished jobs....
  >> 1357    make[2]: *** [parseAPI/CMakeFiles/parseAPI.dir/build.make:742: parseAPI/CMakeFiles/parseAPI.dir/__/dataflowAPI/rose/util/Assert.C.o] Error 1
     1358    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/ThunkData.h:4:0,
     1359                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/IndirectAnalyzer.h:4,
     1360                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/IndirectAnalyzer.C:2:
     1361    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFG.h:404:15: warning: 'Dyninst::ParseAPI::Pa
             rser' is already a friend of 'Dyninst::ParseAPI::Block'
     1362      friend class Parser;
     1363                   ^

     ...

     1442    /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/intel-tbb-2019.2-sfwpny35v2preom52ow3sghnemo52l25/include/tbb/machine/mac_ppc.h: In funct
             ion 'int64_t __TBB_machine_cmpswp8(volatile void*, int64_t, int64_t)':
     1443    /usr/global/tools/openspeedshop/oss-dev/spack/opt/spack/linux-rhel7-ppc64le/gcc-4.9.3/intel-tbb-2019.2-sfwpny35v2preom52ow3sghnemo52l25/include/tbb/machine/mac_ppc.h:116:46: w
             arning: cast from type 'volatile void*' to type 'int64_t* {aka long int*}' casts away qualifiers [-Wcast-qual]
     1444                              , "+m"(* (int64_t*) ptr)        /* redundant with "memory" */
     1445                                                  ^
     1446    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/semantics/BaseSem
             antics2.C:7:0:
     1447    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/semantics/../util/Message.h: In member 
             function 'bool Sawyer::Message::ColorSpec::isDefault() const':
  >> 1448    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/dataflowAPI/rose/semantics/../util/Message.h:389:96: err
             or: cannot convert 'boost::logic::tribool' to 'bool' in return
     1449         bool isDefault() const { return COLOR_DEFAULT==foreground && COLOR_DEFAULT==background && !bold; }
     1450                                                                                                    ^
     1451    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/BoundFactData.h:9:0,
     1452                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/BoundFactData.C:2:
     1453    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFG.h:404:15: warning: 'Dyninst::ParseAPI::Pa
             rser' is already a friend of 'Dyninst::ParseAPI::Block'
     1454      friend class Parser;

     ...

     1768    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFGFactory.h:36:0,
     1769                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CodeObject.h:40,
     1770                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/Function.C:34:
     1771    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFG.h:708:21: warning: unused parameter 'code
             reg' [-Wunused-parameter]
     1772             CodeRegion *codereg) const
     1773                         ^
  >> 1774    make[2]: *** [parseAPI/CMakeFiles/parseAPI.dir/build.make:755: parseAPI/CMakeFiles/parseAPI.dir/__/dataflowAPI/rose/util/Message.C.o] Error 1
     1775    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFGFactory.h:36:0,
     1776                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CodeObject.h:40,
     1777                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/InstructionAdapter.h:36
             ,
     1778                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/IA_IAPI.h:37,
     1779                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/Parser.h:43,
     1780                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/ParseData.C:30:

     ...

     2036                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/IA_aarch64.h:36,
     2037                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/IA_aarch64.C:31:
     2038    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFG.h: At global scope:
     2039    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFG.h:404:15: warning: 'Dyninst::ParseAPI::Pa
             rser' is already a friend of 'Dyninst::ParseAPI::Block'
     2040      friend class Parser;
     2041                   ^
  >> 2042    make[2]: *** [parseAPI/CMakeFiles/parseAPI.dir/build.make:885: parseAPI/CMakeFiles/parseAPI.dir/__/dataflowAPI/rose/semantics/BaseSemantics2.C.o] Error 1
     2043    In file included from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CFGFactory.h:36:0,
     2044                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/CodeObject.h:40,
     2045                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/h/InstructionAdapter.h:36
             ,
     2046                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/IA_IAPI.h:37,
     2047                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/IA_power.h:36,
     2048                     from /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/IA_power.C:32:

     ...

     2231                 ^
     2232    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/Parser.C: At global scope:
     2233    /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/parseAPI/src/Parser.C:573:1: warning: 'void InsertFrames
             (LockFreeQueueItem<Dyninst::ParseAPI::ParseFrame*>*, LockFreeQueue<Dyninst::ParseAPI::ParseFrame*>*)' defined but not used [-Wunused-function]
     2234     InsertFrames
     2235     ^
     2236    make[2]: Leaving directory '/tmp/jeg/spack-stage/spack-stage-ZsG3gg/dyninst/spack-build'
  >> 2237    make[1]: *** [CMakeFiles/Makefile2:1090: parseAPI/CMakeFiles/parseAPI.dir/all] Error 2
     2238    make[1]: Leaving directory '/tmp/jeg/spack-stage/spack-stage-ZsG3gg/dyninst/spack-build'
     2239    make: *** [Makefile:133: all] Error 2

See build log for details:
  /usr/global/tools/openspeedshop/oss-dev/spack/var/spack/stage/dyninst-develop-tupzvz4ltcqmluz5h7a7bwcyagpjbh4h/dyninst/spack-build.out
```

**Comment by jgalarowicz** [comment]
No, but spack is picking the newest version of boost.   Our install-tool 
build tool is set to use 1.67 and that works, but the spack dyninst 
package.py file will choose the latest version of boost.

**Comment by mwkrentel** [comment]
I've submitted PR #541 with a patch to fix this.

The problem was g++ barfing on some implicit conversions from
boost::tribool to bool.  I'm not sure why this fails in the first
place.  Tribool clearly tries to allow seemless integration of tribool
with bool.  I can only guess the implementation is flawed in some way.

Anyway, the solution is simple: add an explicit cast to bool in a few
places.  "Safe and effective."  :-)

**Comment by hainest** [comment]
Fixed in 54a2debd94c3e8df3b6f1e579f1db02dae61cbb7
